tangos/input_handlers/ramsesHOP.py

```diff
@@ -0,0 +1,224 @@
+        return property_list
+
+    @staticmethod
+    def _resolve_units(value):
+        import pynbody
+        if (pynbody.units.is_unit(value)):
+            return float(value)
+        else:
+            return value
+
+    def _get_map_child_subhalos(self, ts_extension):
+        h = self._construct_halo_cat(ts_extension, 'halo')
+        halo_children = {}
+        for halo_i in range(1, len(h)+1):  # AdaptaHOP catalogues start at 1
+            halo_props = h[halo_i].properties
+
+            if halo_props['host_id'] != halo_i: # If halo isn't its own host, it is a subhalo
+                parent = halo_props['host_id']
+                if parent not in halo_children:
+                    halo_children[parent] = []
+                halo_children[parent].append(halo_i)
+        return halo_children
+
+    def iterate_object_properties_for_timestep(self, ts_extension, object_typetag, property_names, index_parent=True):
+        h = self._construct_halo_cat(ts_extension, object_typetag)
+        h._index_parent = index_parent
+
+        if "child" in property_names:
+            # Construct the mapping between parent and subhalos
+            map_child_parent = self._get_map_child_subhalos(ts_extension)
+
+        for halo_i in range(1, len(h)+1):  # AdaptaHOP catalogues start at 1
+
+            # Tangos expects data to have a finder offset, and a finder id following the stat file logic
+            # I think these are irrelevant for AdaptaHOP catalogues which are derived directly from pynbody
+            # Putting the finder ID twice seems to produce consistent results
+            all_data = [halo_i, halo_i]
+
+            adaptahop_halo = h[halo_i]
+
+            if not self._select_halo(adaptahop_halo, min_halo_particles=0):
+                continue
+            precalculated_properties = h[halo_i].properties
+            adaptahop_halo.physical_units() # make sure all units are physical before storing to database
+
+            # Loop over all properties we wish to import
+            for k in property_names:
+                if k in self._include_additional_properties_derived_from_adaptahop():
+                    if k == "parent":
+                        data = proxy_object.IncompleteProxyObjectFromFinderId(precalculated_properties['host_id'], 'halo')
+                    if k == "child":
+                        # Determine whether halo has childs and create halo objects to it
+                        try:
+                            list_of_child = map_child_parent[halo_i]
+                            data = [proxy_object.IncompleteProxyObjectFromFinderId(data_i, 'halo') for data_i in list_of_child]
+                        except KeyError:
+                            data = None
+
+                    # Avoid naming confusions with already defined PynbodyProperties
+                    if k == "shrink_center": data = (adaptahop_halo.properties['pos']).view(np.ndarray)
```

Comment:
Is this definitely the same thing as a `shrink_center`? I don't know how adaptahop defines its halo positions. If it's basically the same, fine; otherwise shouldn't it be called something else to avoid confusion about the reliability of these centres?

Reply:
This is not the same _in general_ as AdaptaHOP has different recipes to compute the centre, one of them being using a shrinking sphere approach. Unfortunately, as far as I can tell, there is no way of knowing what method has been used for centring.

Reply:
I see, I thought the `shrink_center` was the only option for AdapataHOP.  I guess we could call it an `adaptahop_center` and leave it to users to map it to a shrink_center (or else) within their own property module.